MAINT: Correct and update polynomial package documentation.

jeffyancey opened the pull request:
While surveying the polynomial package I attempted to update `Examples`, `See Also`, and make other miscellaneous changes I thought necessary.

I I believe all my changes are in compliance with the official documentation policy, but any feedback is very much appreciated.

mattip committed:
MAINT: test memoryview, improve error message, don't swallow first exception

tylerjereddy committed:
BENCH: Add benchmarks for np.loadtxt reading from CSV format

* NumPy currently has no asv benchmarks for np.loadtxt; now adding
a number of benchmarks for np.loadtxt handling of CSV format files
to prevent performance regressions

mattip committed:
BUG: add tests, special case memoryview(datetime64-scalar)

tempyancey123 committed:
documentation grammer changes

making corrections to documentation grammar while reading through `polynomial/_polybase.py`.

tempyancey123 committed:
documentation grammer changes

making corrections to documentation grammar while reading through `polynomial/_polybase.py`.

tempyancey123 committed:
add examples, "see also" section, and small spelling/formatting fixes

tempyancey123 committed:
add examples, "see also" section, and small spelling/formatting fixes

eric-wieser committed:
Merge pull request #11694 from mattip/master

BUG: pickle and memoryview for datetime64, timedelta64 scalars

jeffyancey commented:
Looks like I didn't push a couple of my reversions. I will do so and respond tomorrow. Thanks for the feedback @eric-wieser

mattip commented on numpy/polynomial/_polybase.py:
```diff
@@ -980,7 +972,7 @@ def basis(cls, deg, domain=None, window=None):
         domain : {None, array_like}, optional
             If given, the array must be of the form ``[beg, end]``, where
-            ``beg`` and ``end`` are the endpoints of the domain. If None is
+            `beg` and `end` are the endpoints of the domain. If None is
```
Just to clarify: in sphinx, as opposed to markdown, text inside a single backtick is a [cross-reference](http://www.sphinx-doc.org/en/stable/tutorial.html#documenting-objects) but without a [domain](http://www.sphinx-doc.org/en/stable/domains.html),  meant to turn into a link using the default domain role. Text inside a double-backtick is formatted as code.

mattip committed:
MAINT: reformat line spacing before test methods

jeffyancey commented on numpy/polynomial/chebyshev.py:
```diff
@@ -1192,32 +1212,33 @@ def chebval2d(x, y, c):
     c : array_like
         Array of coefficients ordered so that the coefficient of the term
-        of multi-degree i,j is contained in ``c[i,j]``. If `c` has
+        of multi-degree i,j is contained in c[i,j]. If `c` has
```
I agree. Will revert.

charris committed:
Merge pull request #11784 from mattip/pep8-test-methods

MAINT: reformat line spacing before test methods

eric-wieser commented on numpy/polynomial/legendre.py:
```diff
@@ -1513,6 +1551,13 @@ def legfit(x, y, deg, rcond=None, full=False, w=None):
     Examples
     --------
+    >>> from numpy.polynomial import legendre as L
+    >>> c = (1,2,3)
```
Ah, that's a good reason to leave it like this. Let's aim for consistency first, and then in another CR we can always overhaul every single example

jeffyancey replied:
I'd like to arrive at a consensus on which of the two example approaches I've seen throughout this directory is preferable:

A:
```
>>> from numpy.polynomial.hermite_e import hermemulx
>>> hermemulx([1, 2, 3])
array([ 2.,  7.,  2.,  3.])
```
B:
```
>>> from numpy.polynomial import chebyshev as C
>>> c = (1,2,3)
>>> C.chebmulx(c)
array([ 1. ,  2.5,  1. ,  1.5])
```

jeffyancey replied:
Once we have reached a consensus, I will make a second PR to go through and have all examples conform, if it is desirable.

tempyancey123 committed:
bevy of changes

1.revert unnecessary/erroneous changes
2. single ticks to double ticks for variables
3. correct spacing on some example output.

tempyancey123 committed:
bevy of changes

1.revert unnecessary/erroneous changes
2. single ticks to double ticks for variables
3. correct spacing on some example output.

tempyancey123 committed:
return accidenally removed `raises` section from ABCPolyBase.convert.

tempyancey123 committed:
return accidenally removed `raises` section from ABCPolyBase.convert.

tempyancey123 committed:
copy-paste added a space. removing.

fml.

tempyancey123 committed:
copy-paste added a space. removing.

fml.

jeffyancey commented:
Reversions have gone through.

jeffyancey commented on numpy/polynomial/chebyshev.py:
```diff
@@ -1156,6 +1167,15 @@ def chebval(x, c, tensor=True):
+    >>> c2 = (3,4)
+    >>> C.chebval((c1, c2), coef)
+    array([[   6.,   26.],
+           [  58.,  102.]])
```
I am on version 1.15.0. I'll doublecheck that is the only version I have installed and that jupyter is 1) using it and 2) not doing anything funny to my output. Thanks for alerting me to this.

charris committed:
Merge pull request #11422 from tylerjereddy/loadtxt_csv_asv_bench

BENCH: Add benchmarks for np.loadtxt reading from CSV format files

tempyancey123 committed:
Merge branch 'polynomial-documentation' of https://github.com/jeffyancey/numpy into polynomial-documentation

jeffyancey commented:
looks like I added all the newly merged commits to my PR accidentally. Sorry all. My changes are  limited to `numpy/polynomial` files.

charris commented:
Might try a rebase on master. If all else fails, google `git reflog` which will allow you to reset git to a previous state.

jeffyancey commented:
whoops, slipped my mind to come back and fix this. Will be done shortly.

jeffyancey commented:
closing and repopening in a clean PR.
